jsHelper/spicetifyWrapper.js

```diff
@@ -1248,481 +1269,376 @@ Spicetify.SVGIcons = {
 	return document.head.appendChild(fontStyle);
 })();
 
-class _HTMLContextMenuItem extends HTMLLIElement {
-	constructor({ name, disabled = false, icon = undefined, trailingIcon = undefined, divider = false }) {
-		super();
-		this.name = name;
-		this.icon = icon;
-		this.trailingIcon = trailingIcon;
-		this.disabled = disabled;
-		this.divider = divider;
-		this.classList.add("main-contextMenu-menuItem");
-	}
-
-	render() {
-		const parseIcon = icon => {
-			if (icon && Spicetify.SVGIcons[icon]) {
-				return `<svg height="16" width="16" viewBox="0 0 16 16" fill="currentColor">${Spicetify.SVGIcons[icon]}</svg>`;
-			}
-			return icon || "";
-		};
-
-		this.innerHTML = `
-		<button class="main-contextMenu-menuItemButton ${this.disabled ? "main-contextMenu-disabled" : ""} ${
-			this.divider ? "main-contextMenu-dividerAfter" : ""
-		}">
-			${parseIcon(this.icon)}
-			<span class="ellipsis-one-line main-type-mesto main-contextMenu-menuItemLabel">${this.name}</span>
-			${parseIcon(this.trailingIcon)}
-		</button>`;
-	}
+Spicetify.ContextMenuV2 = (() => {
+	const registeredItems = new Map();
 
-	connectedCallback() {
-		if (!this.rendered) {
-			this.render();
-			this.rendered = true;
+	function parseProps(props) {
+		if (!props) return;
+		let uris = [];
+		let uids;
+		let contextUri;
+		if (props.uris) {
+			uris = props.uris;
+		} else if (props.uri) {
+			uris = [props.uri];
+		} else if (props.item?.uri) {
+			uris = [props.item.uri];
+		} else if (props.reference?.uri) {
+			uris = [props.reference.uri];
+		} else {
+			return;
+		}
```

Comment:
Hate this as is, can we rewrite this whole function into these bits instead?
```suggestion
		const resolvedUri = props.uri || props.item?.uri || props.reference?.uri;
		const uris = props.uris || (resolvedUri ? [resolvedUri] : null);
		if (!uris) return;
```

Reply:
I'm not the original author of this function but I agree that it's not very readable as is. Though I have to ask, is there a specific reason for using the logical or over the nullish coalescing operator?

Reply:
I think they have the same function here to be honest, both would result in the same functionality, although I feel like the logical OR is more commonly used, so it is more understandable.

Reply:
It's kind of odd not to pair `?.` with `??`, plus it's already been 4 years since the introduction of `??`. But if you don't want to use it that's not big of a deal

Reply:
we should use `??` lol